src/pykeen/models/unimodal/node_piece.py

```diff
@@ -150,29 +151,35 @@ def __init__(
         # Create an MLP for string aggregation
         if aggregation == "mlp":
             aggregation = _ConcatMLP(
-                num_tokens=num_tokens,
+                num_tokens=num_tokens if isinstance(num_tokens, int) else sum(num_tokens),
                 embedding_dim=embedding_dim,
             )
 
         # always create representations for normal and inverse relations and padding
         relation_representations = embedding_specification.make(
             num_embeddings=2 * triples_factory.real_num_relations + 1,
         )
-        # pre-resolve tokenizer to select token representations
-        tokenizer = tokenizer_resolver.lookup(tokenizer)
-        tokenizer_cls = tokenizer.__class__ if isinstance(tokenizer, Tokenizer) else tokenizer
-        # use relation representations for relation tokenizer
-        token_representation: Union[EmbeddingSpecification, RepresentationModule]
-        if tokenizer_cls is RelationTokenizer:
-            token_representation = relation_representations
-        else:
-            # otherwise create new representations
-            token_representation = embedding_specification
+
+        # prepare arguments for the NodePiece Representation
+        representations = []
+        for tkn in tokenizers:
+            # pre-resolve tokenizer to select token representations
+            tokenizer = tokenizer_resolver.lookup(tkn)
+            tokenizer_cls = tokenizer.__class__ if isinstance(tokenizer, Tokenizer) else tokenizer
```

Comment:
@cthoyt this could be moved into class-resolver (i.e. a method which always returns a class, even if an instance is passed)

Reply:
-> https://github.com/cthoyt/class-resolver/issues/18